_stbt/pylint_plugin.py

```diff
@@ -80,6 +86,30 @@ def visit_callfunc(self, node):
                 else:
                     self.add_message('E7003', node=node, args=arg.as_string())
 
+        if _in_frameobject(node):
+            for funcdef in node.func.infer():
+                if (isinstance(funcdef, FunctionDef) and
+                        funcdef != YES and
+                        "frame" in funcdef.argnames()):
+                    index = funcdef.argnames().index("frame")
+                    if node.keywords:
+                        kwargs = [k.arg for k in node.keywords]
+                    else:
+                        kwargs = []
+                    if len(node.args) <= index and "frame" not in kwargs:
```

Comment:
This is elegant.  I like how you don't have to hard-code the list of functions to check against.

On a slightly related note I wonder if df9009eb6f6f8ffb1ebfc20e2e7bc5a850574667 went too far.  Perhaps `match()` should take a `dut=` parameter rather than be a member of `DeviceUnderTest`.  We could pare `DeviceUnderTest` back.